src/main/java/org/fenixedu/academic/domain/student/Registration.java

```diff
@@ -1441,8 +1441,8 @@ private boolean enroledShiftsContainsShiftWithSameExecutionCourseAndShiftType(fi
     }
 
     final public Set<SchoolClass> getSchoolClassesToEnrolBy(final ExecutionCourse executionCourse) {
-        Set<SchoolClass> schoolClasses =
-                executionCourse.getSchoolClassesBy(getActiveStudentCurricularPlan().getDegreeCurricularPlan());
+        Set<SchoolClass> schoolClasses = getActiveStudentCurricularPlan() != null ? executionCourse
```

Comment:
Avoid calling getActiveStudentCurricularPlan() multiple times in the same method.
Invoking the method multiple times results in a performance penalty that should be avoided.